lib/vm/opFns.js

```diff
@@ -91,7 +90,16 @@ module.exports = {
     if (exponent.isZero()) {
       return new BN(1)
     } else {
-      var bytes = 1 + logTable(exponent)
+      let bytesAdd = 32
+      for (let i = 1; i <= 32; i++) {
+        // e.g. 1 -> 0100, 8 -> 010000000000000000
+        let expCmp = new BN('01' + new Array(i * 2).fill(0).join(''), 16)
+        if (exponent.cmp(expCmp) === -1) {
+          bytesAdd = i - 1
+          break
+        }
+      }
+      var bytes = 1 + bytesAdd
       subGas(runState, new BN(bytes).muln(runState._common.param('gasPrices', 'expByte')))
 
       var m = BN.red(utils.TWO_POW256)
```

Comment:
Side note: I'm also wondering if there's a particular reason for using `BN.red` here? do you happen to know if it has better performance than simply doing `base.pow(exponent)`?

Reply:
@axic is a strong proponent of reduction-context based calculation, see also this open PR https://github.com/ethereumjs/ethereumjs-vm/issues/214 using this for the ``MODEXP`` precompile.

I have not enough experiences on this myself, I think main reason is performance optimization, but I can't judge if this applies here. Will leave for now.